routes/infrastructure/construct.py

```diff
@@ -0,0 +1,95 @@
+"""CDK Construct for a Cloudfront Distribution."""
+from typing import Optional
+from urllib.parse import urlparse
+
+from aws_cdk import CfnOutput, Duration, Stack
+from aws_cdk import aws_certificatemanager as certificatemanager
+from aws_cdk import aws_cloudfront as cf
+from aws_cdk import aws_cloudfront_origins as origins
+from aws_cdk import aws_s3 as s3
+from constructs import Construct
+
+from .config import veda_route_settings
+
+
+class CloudfrontDistributionConstruct(Construct):
+    """CDK Construct for a Cloudfront Distribution."""
+
+    def __init__(
+        self,
+        scope: Construct,
+        construct_id: str,
+        raster_api_id: str,
+        stac_api_id: str,
+        region: Optional[str],
+        **kwargs,
+    ) -> None:
+        """."""
+        super().__init__(scope, construct_id)
+
+        stack_name = Stack.of(self).stack_name
+
+        s3Bucket = s3.Bucket.from_bucket_name(
+            self,
+            "stac-browser-bucket",
+            bucket_name=veda_route_settings.stac_browser_bucket,
+        )
+
+        no_cache_policy = cf.CachePolicy(
+            self,
+            "no-cache-policy",
+            default_ttl=Duration.seconds(0),
+            min_ttl=Duration.seconds(0),
+            max_ttl=Duration.seconds(0),
+        )
+
+        # Certificate must be in zone us-east-1
+        domain_cert = (
+            certificatemanager.Certificate.from_certificate_arn(
+                self, "domainCert", veda_route_settings.cert_arn
+            )
+            if veda_route_settings.cert_arn
+            else None
+        )
+
+        self.distribution = cf.Distribution(
```

Comment:
Am I right that with this set of behaviors, the mapping is going to be, assuming the domain is `dev-stac.delta-backend.com` :  

-  `https://dev-stac.delta-backend.com` -> browser
- `https://dev-stac.delta-backend.com/api/stac` STAC API
- `https://dev-stac.delta-backend.com/api/raster` raster API